lib/lambda_ethereum_consensus/p2p/blob_downloader.ex

```diff
@@ -0,0 +1,116 @@
+defmodule LambdaEthereumConsensus.P2P.BlobDownloader do
+  @moduledoc """
+  This module requests blocks from peers.
+  """
+  require Logger
+
+  alias LambdaEthereumConsensus.P2P.ReqResp
+  alias LambdaEthereumConsensus.{Libp2pPort, P2P}
+  alias Types.BlobSidecar
+
+  @blobs_by_range_protocol_id "/eth2/beacon_chain/req/blob_sidecars_by_range/1/ssz_snappy"
+  @blobs_by_root_protocol_id "/eth2/beacon_chain/req/blob_sidecars_by_root/1/ssz_snappy"
+
+  # Requests to peers might fail for various reasons,
+  # for example they might not support the protocol or might not reply
+  # so we want to try again with a different peer
+  @default_retries 5
+
+  @spec request_blobs_by_range(Types.slot(), non_neg_integer(), non_neg_integer()) ::
+          {:ok, [BlobSidecar.t()]} | {:error, any()}
+  def request_blobs_by_range(slot, count, retries \\ @default_retries)
+
+  def request_blobs_by_range(_slot, 0, _retries), do: {:ok, []}
+
+  def request_blobs_by_range(slot, count, retries) do
+    Logger.debug("Requesting blobs", slot: slot)
+
+    # TODO: handle no-peers asynchronously?
+    peer_id = get_some_peer()
+
+    # NOTE: BeaconBlocksByRangeRequest == BlobSidecarsByRangeRequest
+    request =
+      %Types.BeaconBlocksByRangeRequest{start_slot: slot, count: count}
+      |> ReqResp.encode_request()
+
+    with {:ok, response} <-
+           Libp2pPort.send_request(peer_id, @blobs_by_range_protocol_id, request),
+         {:ok, blobs} <- ReqResp.decode_response(response, BlobSidecar),
+         :ok <- verify_batch(blobs, slot, count) do
+      {:ok, blobs}
+    else
+      {:error, reason} ->
+        P2P.Peerbook.penalize_peer(peer_id)
```

Comment:
eventually we want to penalize based on the error, right? Or are all errors penalizable?

Reply:
Maybe. We can revisit this later.